tests/unit/st_assert.hpp

```diff
@@ -0,0 +1,54 @@
+// st_assert.hpp - Assert function for tests
+// Copyright (C) 2024 Hyland B. <me@ow.swag.toys>
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+#pragma once
+#include <iostream>
+#include <optional>
+#include <cstdlib>
+#include <string_view>
+
+namespace {
+
+#define ST_ASSERT(name, expr) st_assert(__FILE__, __LINE__, name, expr)
+
+// TODO Make macros for these assert functions to also pass a line number
+void st_assert(const std::string_view filename,
```

Comment:
I think string_view should be passed by reference (even though it's not as expensive as std::string)

Reply:
`string_view` provides read-only access to an existing string, it does not represent the string itself. Therefore you can pass it as value

Reply:
I recently got a new color printer, so, color me surprised. I didn't know that.

Reply:
Actually I kind of agree with Tobbi on one thing, I dont need to use string view here. I think I can just use a good old char* since im just printing the value out.